plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelAction.java

```diff
@@ -188,24 +183,35 @@ void processUndeployModelResponseAndUpdate(
                             "updated model state as undeployed for : {}",
                             Arrays.toString(actualRemovedNodesMap.keySet().toArray(new String[0]))
                         );
-                }, e -> { log.error("Failed to update model state as undeployed", e); });
+                    countDownLatch.countDown();
+                }, e -> {
+                    log.error("Failed to update model state as undeployed", e);
+                    countDownLatch.countDown();
+                });
                 client.bulk(bulkRequest, ActionListener.runAfter(actionListener, () -> {
                     syncUpUndeployedModels(syncUpRequest);
-                    listener.onResponse(undeployModelNodesResponse);
+                    context.restore();
                 }));
             } else {
                 syncUpUndeployedModels(syncUpRequest);
-                listener.onResponse(undeployModelNodesResponse);
+                context.restore();
+                countDownLatch.countDown();
+            }
+        }
+        if (countDownLatch.getCount() != 0) {
+            try {
+                boolean success = countDownLatch.await(1000, TimeUnit.MILLISECONDS);
```

Comment:
Is this 1 second too large? In most cases the update index happen pretty fast.

Reply:
This is a `at most` value, if the update index happens within 1s, the await will break immediately after it's done. If update takes more than 1s, then await waits at most 1s to break.

Reply:
This will block current thread, which not recommended. We saw issues several times for such blocking thread code

Reply:
If you're not blocking event loop thread, it's safe.